packages/flutter_test/lib/src/binding.dart

```diff
@@ -494,9 +494,25 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   ///
   /// When [handlePointerEvent] is called directly, [pointerEventSource]
   /// is [TestBindingEventSource.device].
+  ///
+  /// This default means that pointer events triggered by the [WidgetController]
+  /// will result in actual interactions with the UI, but other pointer events
+  /// such as those from physical taps will be delegated to
+  /// [deviceEventDispatcher] *instead*. See also
+  /// [shouldPropagateDevicePointerEvents] if this is undesired.
   TestBindingEventSource get pointerEventSource => _pointerEventSource;
   TestBindingEventSource _pointerEventSource = TestBindingEventSource.device;
 
+  /// Whether pointer events from [TestBindingEventSource.device] will be
+  /// propagated to the framework, or dropped.
+  ///
+  /// Setting this can be useful to interact with the app in some other way
+  /// besides through the [WidgetController], such as with `adb shell input tap`
+  /// on Android.
+  ///
+  /// See also [pointerEventSource].
+  bool shouldPropagateDevicePointerEvents = false;
```

Comment:
If we land this, we'd need ways to make sure it gets reset after tests.

The frameworks normal solution here would be to subclass the bindings. Is that still not possible in google's testing environment?

Reply:
We can subclass the binding in Google3, just that it adds a bit of complexity and I don't think it's necessary if this is more formally supported through a setting like this.

Is it important for us to reset the value though? I think users should be free to change this value wherever as some sort of "global" - whether it is in a `testWidgets()` or at the top of their `main()`, depending on their use case.

Reply:
Done in the other comment below